ivy_tests/test_ivy/helpers/hypothesis_helpers/dtype_helpers.py

```diff
@@ -448,3 +478,154 @@ def bound_dtype_bits(d):
             )
             and (min_x > 0 or not ivy_backend.is_uint_dtype(dtype))
         )
+
+
+possible_castable_dtypes = {
```

Comment:
as mentioned before, we shouldn't need this code as the casting modes should be able to cover this behaviour. If there's issues with the casting modes, we should be fixing the casting modes instead (cc @RickSanchezStoic)

Reply:
Hey @vedpatwardhan  I discussed it with @RickSanchezStoic. he suggested me to use different logic. Also, there's couple of restrictions with casting modes predictability but here, we even need the castable dtype for generating data arrays. like If the casting mode will handle the dtype under the hood than how we can restrict data to that dtype, we don't know which dtype it'll casted to. I think if @RickSanchezStoic confirm this thing again here, It'll be helpful. thanks!

Reply:
Happy to know what you think @RickSanchezStoic, but I think if we still need to duplicate the casting logic such that we do the casting before passing the input to the function instead of using the casting modes kinda defeats the purpose of the casting modes because then we won't be able to test if the casting modes are working as expected. I think fixing the casting modes to capture all cases which lead to this duplication should be the way to go 😅